File: codecheaf/REPLESX.cpp
#include <bits/stdc++.h>
using namespace std;
#define mod 1000000007

int findIndex(int a[], int N, int X, int p)
{
    int min = 100000000, minp = 0;
    for (int i = 0; i < N; i++)
    {
        if (a[i] == X)
        {
            if (abs(p - i) < min)
            {
                min = abs(p - i);
                minp = i;
            }
        }
    }

    return minp;
}
void subMain()
{
    int N, X, p, k;
    cin >> N >> X >> p >> k;
    int a[N];
    for (int i = 0; i < N; i++)
    {
        cin >> a[i];
    }
    sort(a, a + N);
    int index = findIndex(a, N, X, p);
    int count = 0;
    if (a[index] != X)
    {
        a[k - 1] = X;
        sort(a, a + N);
        count++;
    }
    if (a[p - 1] == X)
    {
        cout << 0 + count << endl;
        return;
    }
    if (p < k && a[p - 1] < X)
    {
        cout << -1 << endl;
        return;
    }
    if (p > k && a[p - 1] > X)
    {
        cout << -1 << endl;
        return;
    }
    index = findIndex(a, N, X, p) + 1;
    cout << abs(p - index) + count << endl;
}

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        subMain();
    }
}

File: codecheaf/REPLESX.py
import sys
sys.stdin = open("input.txt", "r")

def findIndex(L, N, X, p):
    min = 100000000
    minp = 0
    for i in range(N):
        if L[i] == X:
            if abs(p -i) < min:
                min = abs(p-i)
                minp = i
    return minp



def subMain():
    for _ in range(int(input())):
        N, X, p, k = map(int, input().split())
        L = list(map(int, input().split()))
        L.sort()
        index = findIndex(L, N, X, p) 
        count = 0
        if L[index] != X:
           L[k - 1] = X
           L.sort()
           count += 1

        if L[p -1] == X:
           print(0 + count)
           return
        if (p < k) and (L[p-1] < X):
           print(-1)
           return
        if (p > k) and (L[p-1] > X):
            print(-1)
            return   
        ind = findIndex(L, N, X, p) + 1
        print(abs(p-ind) + count)   

if __name__ == "__main__":
    subMain()
